crates/supervisor/core/src/syncnode/task.rs

```diff
@@ -276,6 +271,43 @@ where
         }
     }
 
+    async fn check_node_consistency(
+        &self,
+        derived_ref_pair: &DerivedRefPair,
+        next_l1_block: &Block,
+    ) -> Result<(), ManagedEventTaskError> {
+        if next_l1_block.header.parent_hash != derived_ref_pair.source.hash {
+            error!(target: "managed_event_task", "L1 Block parent hash mismatch");
+            self.handle_reset("l1 block parent hash mismatch").await;
+            Err(ManagedEventTaskError::BlockHashMismatch {
+                current: derived_ref_pair.source.hash,
+                parent: next_l1_block.header.parent_hash,
+            })?
+        }
+
+        // check if the derived block is already stored and is consistent with the incoming derived
+        // block
+        let derived_block = derived_ref_pair.derived;
+
+        match self.db_provider.get_block(derived_block.number) {
```

Comment:
This looks incorrect. Here, we should fetch the latest derived block from the storage.